### PR TITLE
Don't create engine via import

### PR DESCRIPTION
**Pull request by gabriel**
The import was constructing the engine.

### CODE REVIEW

**Comment by gabriel** [comment]
@keybase/react-hackers

File: shared/engine/index.js [hunk 1]
```diff
@@ -432,4 +432,11 @@ const cancelError = {
   desc: 'Canceling RPC',
 }
 
-export default new Engine()
```
Comment (gabriel) [inline shared/engine/index.js]: This is the main change.

**Comment by chrisnojima** [comment]
What is the purpose of this change

**Comment by chrisnojima** [comment]
Also I have refactored engine so let's not merge this

**Comment by gabriel** [comment]
Imports shouldn't be constructing stuff as a general rule right?

For example, we can't log anything on engine create because the log setup happens after the engine import. For example, tt seemed awkward that an import would cause a connection to happen. Also you are relying on imports to enforce the singleton pattern which is probably ok, but... you know.

I thought your engine fix was merged. I can nuke this and re-apply it after that gets merged in?

**Comment by gabriel** [comment]
Also, btw, singletons are an anti pattern. But I'm ok with the 1 singleton per app rule (which I usually call the app instance/singleton). But that's another discussion I suppose.

**Pull request closed without merging**